src/emc/plots/spatial.py

```diff
@@ -0,0 +1,160 @@
+                  crs=ccrs.PlateCarree())
+    ax.set_yticks([10, 30, 50, 70], crs=ccrs.PlateCarree())
+    lon_formatter = LongitudeFormatter(zero_direction_label=False)
+    lat_formatter = LatitudeFormatter()
+    ax.xaxis.set_major_formatter(lon_formatter)
+    ax.yaxis.set_major_formatter(lat_formatter)
+
+    return fig, ax
+
+
+def _conus():
+    """
+    Return fig, ax on CONUS domain.
+    """
+    fig = plt.figure(figsize=(10, 8))
+    ax = fig.add_subplot(1, 1, 1,
+                         projection=ccrs.PlateCarree(central_longitude=0))
+    ax.add_feature(cfeature.GSHHSFeature(scale='auto'))
+    ax.set_extent([-130, -60, 10, 60])
+    ax.set_xticks([-130, -120, -110, -100, -90, -80, -70, -60],
+                  crs=ccrs.PlateCarree())
+    ax.set_yticks([10, 20, 30, 40, 50, 60], crs=ccrs.PlateCarree())
+    lon_formatter = LongitudeFormatter(zero_direction_label=False)
+    lat_formatter = LatitudeFormatter()
+    ax.xaxis.set_major_formatter(lon_formatter)
+    ax.yaxis.set_major_formatter(lat_formatter)
+
+    return fig, ax
+
+
+def _spatial_scatter(fig, ax, longitude, latitude, data, plotopts):
+    """
+    Plot 1-dimensional data on a map as scatter.
+    """
+    cs = plt.scatter(latitude, longitude, c=data, s=plotopts['marker size'],
+                     vmin=plotopts['vmin'], vmax=plotopts['vmax'],
+                     cmap=plotopts['cmap'], transform=ccrs.PlateCarree())
+
+    cb = plt.colorbar(cs, shrink=0.5, pad=.03, extend='both')
+    cb.set_label(plotopts['cbar label'])
+
+    plt.grid(plotopts['grid'])
+    plt.title(plotopts['title'], loc='left')
+    plt.title(plotopts['cycle'], loc='right', fontweight='semibold')
+
+    return fig
+
+
+def _spatial_pcolormesh(fig, ax, longitude, latitude, data, plotopts):
+    """
+    Plot 2-dimensional data on a map as pcolormesh.
+    """
+    cs = plt.pcolormesh(lon, lat, data, cmap=plotopts['cmap'],
+                        vmin=plotopts['vmin'], vmax=plotopts['vmax'],
+                        transform=ccrs.PlateCarree())
+
+    cb = plt.colorbar(cs, shrink=0.5, pad=.03, extend='both')
+    cb.set_label(plotopts['cbar label'])
+
+    plt.grid(plotopts['grid'])
+    plt.title(plotopts['title'], loc='left')
+    plt.title(plotopts['cycle'], loc='right', fontweight='semibold')
+
+    return fig
+
+
+def spatial(latitude, longitude, data, domain='global', cmap='viridis',
+            markersize=10, grid=False, vmin=None, vmax=None, title=None,
+            cycle=None, cbar_label=None):
+    """
+    Plot a spatial map of desired data.
+    Paramaters
+    ----------
+    Required:
+    latitude, longitude, data : array type (1D or 2D)
+    domain : domain of map (str)
+
+    Optional:
+    cmap : colormap (str)
+           (https://matplotlib.org/stable/gallery/color/colormap_reference.html)
+    vmin : minumum value of data
+    vmax : maximum value of data
+    title : plot title (str)
+    cycle : cycle of data (str)
+    cbar_label : colorbar label
```

Comment:
we need to work on the syntax of this now before it is too far.
Investigate `pdoc` syntax.